### PR TITLE
Simplify debug exercise

### PR DESCRIPTION
**Pull request by hageboeck**
In addition to the work in #468, I propose to simplify the code. It seemed to be complicated enough that we didn't fix an out-of-bounds access in the solution, so here I propose to take the randomness out of the exercise, and go for a simple reverse.

During the C++ course at JLab, a mentor suggested to add a check for correctness, which is trivial when the collection just has to be reversed.

I did a similar thing for the valgrind part, but I added an off-by-one mistake in the reverse function, and a memory leak in main. That should hopefully be enough for people to find.

Note: Either close #468 and merge the current one, or merge #468 and this gets rebased.

### CODE REVIEW

File: code/debug/debug.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+void swap(int *a, int*b) {
+    int c = *a;
+    *a = *b;
+    *b = c;
+}
+
+void reverse(int *v, unsigned int len) {
+    for (unsigned int i = 0; i < (len + 1) / 2; i++) {
+      const int a = i;
+      const int b = len - 1 - i;
```
Comment (bernhardmgruber) [inline code/debug/debug.cpp:13]: So my wristband convinced you today? :)

**Review by bernhardmgruber — APPROVED** [review]

**Review by bernhardmgruber — CHANGES_REQUESTED** [review]
Please update the `README.md` of the debug exercise as well!

**Review by sponce — APPROVED** [review]

**Review by bernhardmgruber — APPROVED** [review]

File: code/debug/debug.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      v[i] = i;
+    }
+}
+
+int main() {
+    constexpr auto arraySize = 100;
+    int *v = nullptr;
+    // create and reverse the vector of LEN numbers
+    reverse(v, 1000);
+    createAndFillVector(v, arraySize);
```
Comment (pcanal) [inline code/debug/debug.cpp:31]:
```suggestion
    v = createAndFillVector(arraySize);
```

File: code/debug/debug.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      swap(v + a, v + b);
+    }
+}
+
+void createAndFillVector(int*& v, unsigned int len) {
+    v = new int[len];
+    for (unsigned int i = 0; i < len; i++) {
+      v[i] = i;
+    }
+}
```
Comment (pcanal) [inline code/debug/debug.cpp:24]:
```suggestion
int *createAndFillVector(unsigned int len) {
    auto v = new int[len];
    for (unsigned int i = 0; i < len; i++) {
      v[i] = i;
    }
    return v;
}
```

File: code/debug/debug.cpp (hunk 2 above)
Comment (hageboeck) [reply code/debug/debug.cpp]: Very good, thanks! It goes in the direction of value semantics, so I will do this transformation also with the other three files.

**Pull request merged**